alibi_detect/od/_knn.py

```diff
@@ -0,0 +1,221 @@
+            value. For a list of available aggregators, see :mod:`alibi_detect.od.pytorch.ensemble`.
+        backend
+            Backend used for outlier detection. Defaults to ``'pytorch'``. Options are ``'pytorch'``.
+        device
+            Device type used. The default tries to use the GPU and falls back on CPU if needed.
+            Can be specified by passing either ``'cuda'``, ``'gpu'``, ``'cpu'`` or an instance of
+            ``torch.device``.
+
+        Raises
+        ------
+        ValueError
+            If `k` is an array and `aggregator` is None.
+        NotImplementedError
+            If choice of `backend` is not implemented.
+        """
+        super().__init__()
+
+        backend_str: str = backend.lower()
+        BackendValidator(
+            backend_options={'pytorch': ['pytorch']},
+            construct_name=self.__class__.__name__
+        ).verify_backend(backend_str)
+
+        backend_cls, ensembler_cls = backends[backend]
+        ensembler = None
+
+        if aggregator is None and isinstance(k, (list, np.ndarray, tuple)):
+            raise ValueError('If `k` is a `np.ndarray`, `list` or `tuple`, '
+                             'the `aggregator` argument cannot be ``None``.')
+
+        if isinstance(k, (list, np.ndarray, tuple)):
+            ensembler = ensembler_cls(
+                normalizer=get_normalizer(normalizer),
+                aggregator=get_aggregator(aggregator)
+            )
+
+        self.backend = backend_cls(k, kernel=kernel, ensembler=ensembler, device=device)
+
+        # set metadata
+        self.meta['detector_type'] = 'outlier'
+        self.meta['data_type'] = 'numeric'
+        self.meta['online'] = False
+
+    def fit(self, x_ref: np.ndarray) -> None:
+        """Fit the detector on reference data.
+
+        Parameters
+        ----------
+        x_ref
+            Reference data used to fit the detector.
+        """
+        self.backend.fit(self.backend._to_tensor(x_ref))
+
+    @catch_error('NotFittedError')
+    @catch_error('ThresholdNotInferredError')
+    def score(self, x: np.ndarray) -> np.ndarray:
+        """Score `x` instances using the detector.
+
+        Computes the k nearest neighbor distance/kernel similarity for each instance in `x`. If `k` is a single
+        value then this is the score otherwise if `k` is an array of values then the score is aggregated using
+        the ensembler.
+
+        Parameters
+        ----------
+        x
+            Data to score. The shape of `x` should be `(n_instances, n_features)`.
+
+        Raises
+        ------
+        NotFittedError
+            If called before detector has been fit.
+        ThresholdNotInferredError
+            If k is a list and a threshold was not inferred.
+
+        Returns
+        -------
+        Outlier scores. The shape of the scores is `(n_instances,)`. The higher the score, the more anomalous the \
+        instance.
```

Comment:
Line continuation `\` unnecessary?

Reply:
I we remove autodoc thinks it's two separate return values and formats as alist:

![docs-issue](https://user-images.githubusercontent.com/29690500/225711377-ca5ec1df-004a-45d2-805b-a162abd8ad71.png)

Reply:
Looking at the `alibi` docstrings further, it seems our usual convention is either to do:

```
    Returns
    -------
    param_name1
        Description.
    param_name2
        Description.
```

Or alternatively:

```
    Returns
    -------
    Extended description blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah 
            blah blah blah
```

The indent on the second line appears to have the same effect as your `\`...

Apologies, I think I asked to move away from the first option initially. Although in my defense I think that was more to note that we don't need to specify the return types as in the following (because we use `sphinx-autodoc-typehints`:

```
    Returns
    -------
    rtype1
        Description.
    rtype2
        Description.
```

Regarding the PR, I don't have strong opinions wrt to using indents or `\`...

Reply:
Oh, I've actually fixed a bunch of these in #748. I was aware of it as @RobertSamoilescu highlighted the issue in [alibi](https://github.com/SeldonIO/alibi/blob/master/CONTRIBUTING.md#docstrings). I had thought the same effort to fix them there had been done in detect but it hasn't yet!